tasks/eva_2835/gather_per_species_per_assembly_variant_remapping_counts.py

```diff
@@ -0,0 +1,270 @@
+#Copyright 2022 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Importing relevant packages
+import os
+import yaml
+import argparse
+import xlsxwriter
+from string import ascii_uppercase
+
+
+# Initializing the final output column
+all_columns=[]
+
+# Initializing the list to store all the keys and values for all the files
+all_keys=[]
+all_values=[]
+
+# Initializing the list to store the values of all taxonomy ids and assembly accessions
+all_tax_id=[]
+all_assembly=[]
+
+
+def generate_output(remapping_root_path, output_path):
```

Comment:
From the function name, we would assume that it will only create the output but it does create the entire data structure as well.
I think it makes sense to have a function to create the output but then you should extract the functionalities that retrieve the data to another function.

Reply:
I have created a separate function for generation of output

Reply:
You have simplified it indeed but it still contains both the data retrieval and the generation of the output in the same function.

Reply:
I have now made the relevant changes for  your review